components/AudioPlayer.tsx

```diff
@@ -0,0 +1,388 @@
+
+      return `${returnedMinutes}:${returnedSeconds}`;
+    }
+
+    return "00:00";
+  };
+
+  const togglePlayPause = () => {
+    const prevValue = isPlaying;
+
+    setIsPlaying(!prevValue);
+    if (!prevValue) {
+      audioPlayer.current?.play();
+      animationRef.current = requestAnimationFrame(whilePlaying);
+    } else {
+      audioPlayer.current?.pause();
+      cancelAnimationFrame(animationRef.current);
+    }
+  };
+
+  const whilePlaying = () => {
+    if (audioPlayer.current) {
+      progressBar.current!.value = String(audioPlayer.current.currentTime);
+      changePlayerCurrentTime();
+      animationRef.current = requestAnimationFrame(whilePlaying);
+    }
+  };
+
+  const changeRangeOfProgressBar = () => {
+    audioPlayer.current!.currentTime = Number(progressBar.current?.value);
+    changePlayerCurrentTime();
+  };
+
+  const changeRangeOfVolumeBar = useCallback(() => {
+    audioPlayer.current!.volume = Number(volumeBar.current?.value) / 100;
+    changePlayerVolume();
+  }, []);
+
+  const changePlayerVolume = () => {
+    volumeBar.current?.style.setProperty(
+      "--seek-before-width",
+      `${Number(volumeBar.current.value)}%`
+    );
+    setVolume(Number(volumeBar.current?.value));
+  };
+
+  const changePlayerCurrentTime = () => {
+    progressBar.current?.style.setProperty(
+      "--seek-before-width",
+      `${(Number(progressBar.current.value) / duration) * 100}%`
+    );
+    setCurrentTime(Number(progressBar.current!.value));
+  };
+
+  useEffect(() => {
+    if (!pageInitialized.current) {
+      volumeBar.current!.value = "100";
+      changeRangeOfVolumeBar();
+    }
+
+    if (audioPlayer.current?.duration) {
+      const seconds = Math.floor(audioPlayer.current.duration);
+
+      setDuration(seconds);
+      progressBar.current!.max = String(seconds);
+    }
+
+    if (
+      Math.ceil(audioPlayer.current!.currentTime) ===
+      Number(progressBar.current!.max)
+    ) {
+      setIsPlaying(false);
+    }
+    pageInitialized.current = true;
+  }, [
+    audioPlayer?.current?.onloadedmetadata,
+    audioPlayer.current?.readyState,
+    audioPlayer.current?.currentTime,
+    changeRangeOfVolumeBar,
+  ]);
+
+  return (
+    <Box
+      sx={(theme) => ({
+        position: "relative",
+        width: "100%",
+        height: "53px",
+        display: "flex",
+        alignItems: "center",
+        justifyContent: "space-between",
+        padding: "8px 10px 7px 10px",
+        backgroundColor: theme.palette.mode === "light" ? "#f2f5f7" : "#33393f",
+        borderRadius: "4px",
+        marginBottom: "18px",
+        "& button": {
+          border: "none",
+          cursor: "pointer",
+          p: 0,
+        },
+      })}
+    >
+      <audio
+        ref={audioPlayer}
+        src="https://cdn.simplecast.com/audio/cae8b0eb-d9a9-480d-a652-0defcbe047f4/episodes/af52a99b-88c0-4638-b120-d46e142d06d3/audio/500344fb-2e2b-48af-be86-af6ac341a6da/default_tc.mp3"
```

Comment:
put this value, in a variable something like testAudioSample or something in that way